File: disciple/src/main/java/com/monster/disciple/subscribe/IDiscipleException.java
package com.monster.disciple.subscribe;

import com.monster.disciple.exception.BusinessException;

public interface IDiscipleException {

    /**
     * 网络 异常
     */
    void onBadNetError(Throwable t);

    /**
     * http 异常
     */
    void onServerError(Throwable t);

    /**
     * 业务 异常
     */
    void onBusinessError(BusinessException ex);

    /**
     * 其他 异常
     */
    void onOtherError(Throwable t);
}

File: disciple/src/main/java/com/monster/disciple/subscribe/IDiscipleException.kt
package com.monster.disciple.subscribe

import com.monster.disciple.exception.BusinessException

interface IDiscipleException {

    /**
     * 网络 异常
     */
    fun onBadNetError(t: Throwable)

    /**
     * http 异常
     */
    fun onServerError(t: Throwable)

    /**
     * 业务 异常
     */
    fun onBusinessError(ex: BusinessException)

    /**
     * 其他 异常
     */
    fun onOtherError(t: Throwable)
}